### ISSUE TITLE
Custom Training DataSet

### ISSUE THREAD

**Issue by NitinAggarwal1** [Extreme-classification/ngame#1]
Hi Kunal,

I have been trying to create the training data set for LF-Amazon-131K dataset . I have downloaded the raw data from the following link [https://drive.google.com/file/d/1WuquxCAg8D4lKr-eZXPv4nNw2S2lm7_E/view?usp=sharing](url)
 
 The dataset consists of three files , trn.json.gz , tst.json.gz and lbl.json.gz 

Understanding of the files is as follows:

1.  “trn.json.gz" and “tst.json.gz" have information about the labels in the following format :  
{"**uid**": "B0002QNKW2", "**title**": "Amazon.com: Playtex Women's Cross Your Heart Lightly Lined Seamless Soft Cup Bra #655: Clothing", "**content**": "Style #655.  Our best selling Cross Your Heart bra!  Famous cross your heart frame lifts and separates for more natural curves under clothes.\tClassic; clean lines combine with a smooth; seamless design to make this bra perfect for every day. Seamless cups with feather-light lining give you a smooth; no-show-through look. Convenient front-adjustable straps stretch for A-C cups and add extra support for D sizes with a non-stretch material. Stretch sides and back hug your body for a great fit and comfort all day long.", "**target_ind**": [89490, 89491, 118419, 118424], "**target_rel**": [1.0, 1.0, 1.0, 1.0]}
It includes title , content  as text of the data: For the labels and its score we have target_ind and target_rel  respectively.
 
2. “lbl.json.gz” have information about the labels matching based on indices which are relevant . 89490 ==> refers to index in label file which is relevant . This matches with target_ind that is present in ‘trn.json' and ‘tst.json'.

Steps in Construction of Training Data in NGAME format . 

1. prepare_data.sh 
It outputs the six tokenised files based on the tokeniser specified( Bert-base-uncased) correctly.
 
    - trn_doc_input_ids.npy
    - trn_doc_attention_mask.npy
    - tst_doc_input_ids.npy
    - tst_doc_attention_mask.npy
    - lbl_input_ids.npy
    - lbl_attention_mask.npy

2. For generating the two other files : trn_X_Y.txt and tst_X_Y.txt
I have used the convert_format.pl $data_dir/train.txt $data_dir/trn_X_Xf.txt $data_dir/trn_X_Y.txt  ( **Something does not look right here**)

Can you please confirm if it is correct or there is a different way to output the trn_X_Y.txt and tst_X_Y.txt.  

PS : Just want to make sure the process is exactly followed so that it can be ported on any custom dataset and is useful for others.

**Comment by kunaldahiya** [comment]
Hi

Thanks for checking out NGAME. The steps look correct. 

Reg. {trn, tst}_X_Y.txt: Since, we already had the labels stored in the files used by older methods (please note that train.txt and test.txt contains sparse bag-of-words features and labels). So, we just extracted from there. It is also possible to create trn_X_Y.txt (and tst_X_Y.txt) from the .gz files itself. I can share the script for the same.

**Comment by NitinAggarwal1** [comment]
Got it . Please share the script so that it is better to work directly on raw data so that everyone can use it directly from the raw format itself.

**Comment by kunaldahiya** [comment]
Updated. Please take a look. Data preparation process should be simpler now. Thanks for pointing it out.

**Comment by NitinAggarwal1** [comment]
Closed the issue . As per the understanding there is no need of BOW features for Ngame as encoding happens via a transformer model.
